functions/Find-DbaCommand.ps1

```diff
@@ -0,0 +1,154 @@
+
+Finds all commands searching the entire help for "snapshot", rebuilding the index (good for developers)
+#>
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    Param (
+        [String]$Pattern,
+        [String]$Tag,
+        [String]$Author,
+        [switch]$Rebuild
+    )
+    BEGIN
+    {
+        $tagsRex = ([regex]'(?m)^[\s]{0,15}Tags:(.*)$')
+        $authorRex = ([regex]'(?m)^[\s]{0,15}Author:(.*)$')
+
+        function Get-DbaHelp([String]$commandname) {
+            $thishelp = Get-Help $commandname -Full
+            $thebase = @{}
+            $thebase.CommandName = $commandname
+            $thebase.Name = $thishelp.name
+
+            ## fetch the description
+            $thebase.Description = $thishelp.Description.text
+
+            ## fetch examples
+            $thebase.Examples = $thishelp.Examples | Out-String -Width 120
+
+            ## fetch help link
+            $thebase.Links = ($thishelp.relatedLinks).navigationLink.uri
+
+            ## fetch the synopsis
+            $thebase.Synopsis = $thishelp.Synopsis
+
+            ## store notes
+            $as = $thishelp.alertSet | Out-String -Width 120
+
+            ## fetch the tags
+            $tags = $tagsrex.Match($as).Groups[1].Value
+            if($tags) {
+                $thebase.Tags = $tags.Trim().Split(',')
+            }
+            ## fetch the author
+            $author = $authorRex.Match($as).Groups[1].Value
+            if($author) {
+                $thebase.Author = $author.Trim()
+            }
+
+            [pscustomobject]$thebase
+        }
+
+        function Get-DbaIndex() {
+            $dbamodule = Get-Module dbatools
+            $allcommands = $dbamodule.ExportedCommands
+            $helpcoll = New-Object System.Collections.Generic.List[System.Object]
+            foreach($c in $allcommands.GetEnumerator()) {
+                $x = Get-DbaHelp $c.Key
+                $helpcoll.Add($x)
+            }
+            $dest = Get-ConfigValue -Name 'Path.TagCache' -Fallback "$(Resolve-Path $PSScriptRoot\..)\dbatools-index.json"
+            if ($Pscmdlet.ShouldProcess($dest, "Recreating index"))
+            {
+                $helpcoll | ConvertTo-Json | Out-File $dest
+            }
+        }
+    }
+    PROCESS
+    {
+        $idxfile = Get-ConfigValue -Name 'Path.TagCache' -Fallback "$(Resolve-Path $PSScriptRoot\..)\dbatools-index.json"
+        if(!(Test-Path $idxfile) -or $Rebuild) {
+            Write-Verbose "Rebuilding index into $idxfile"
+            $swrebuild = [system.diagnostics.stopwatch]::startNew()
+            Get-DbaIndex
+            Write-Verbose "Rebuild done in $($swrebuild.Elapsedmilliseconds)ms"
+
+        }
+        $consolidated = Get-Content $idxfile | ConvertFrom-Json
+        $result = $consolidated
+        if($Pattern.length -gt 0) {
+            $result = $result | Where-Object { $_.psobject.properties.value -like "*$Pattern*" }
+        }
+        if($Tag.length -gt 0) {
```

Comment:
Could you do a split on $Tag too and return the commands that have multiple tags?

Find-DbaCommand -Tag copy,user